src/metrics.jl

```diff
@@ -213,12 +213,14 @@ for dist in weightedmetrics
     @eval parameters(d::$dist) = d.weights
 end
 
-result_type(dist::UnionMetrics, a::AbstractArray, b::AbstractArray) =
-    result_type(dist, a, b, parameters(dist))
-result_type(dist::UnionMetrics, a::AbstractArray, b::AbstractArray, ::Nothing) =
-    typeof(_evaluate(dist, oneunit(eltype(a)), oneunit(eltype(b))))
-result_type(dist::UnionMetrics, a::AbstractArray, b::AbstractArray, p) =
-    typeof(_evaluate(dist, oneunit(eltype(a)), oneunit(eltype(b)), oneunit(eltype(p))))
+result_type(dist::UnionMetrics, a::AbstractArray, b::AbstractArray, p=parameters(dist)) =
+    result_type(dist, eltype(a), eltype(b), p)
```

Comment:
For clarity, I'd suggest to remove this method. Let `result_type` with arrays be handled by the fallback, which calls the method below, and further one of the two subsequent ones.
```suggestion
```
Then in the `_evaluate` method, the `p` argument needs to be removed in the `zero(result_type(d, a, b, p))`.

Reply:
Much clearer!